user-data-store-server/src/main/java/com/wultra/security/userdatastore/controller/PhotoController.java

```diff
@@ -123,4 +126,40 @@ public Response deletePhotos(@NotBlank @Size(max = 255) @RequestParam String use
         return new Response();
     }
 
+    /**
+     * Import photos synchronously.
+     *
+     * @param request Photo import request
+     * @return photo import response
+     */
+    @Operation(
+            summary = "Import photos",
+            description = "Import photos synchronously."
+    )
+    @PostMapping("/admin/photos/import")
+    public ObjectResponse<PhotosImportResponse> importPhotos(@Valid @RequestBody final ObjectRequest<PhotosImportRequest> request) {
+        logger.info("action: importPhotos, state: initiated");
+        final PhotosImportResponse response = photoService.importPhotos(request.getRequestObject());
+        logger.info("action: importPhotos, state: succeeded");
+        return new ObjectResponse<>(response);
+    }
+
+    /**
+     * Import photos asynchronously from CSV.
+     *
+     * @param request Photo import from CSV request
+     * @return Response
+     */
+    @Operation(
+            summary = "Import photos from CSV",
+            description = "Import photos asynchronously from CSV."
+    )
+    @PostMapping("/admin/photos/import/csv")
```

Comment:
🤔 A nitpick: I was about to suggest extracting the common path into `@RequestMapping` but I see that we mix admin and non-admin context. Shouldn't we split the controller into two classes?

Reply:
I prefer to group methods related to the same functionality in one controller class.